src/ert/config/queue_config.py

```diff
@@ -108,11 +115,68 @@ def generate_dict(option_list: List[Tuple[str, str]]) -> Dict[str, List[str]]:
 
 queue_memory_options: Mapping[str, List[str]] = {
     "LSF": [],
-    "SLURM": [],
+    "SLURM": ["MEMORY_PER_CPU", "MEMORY"],
     "TORQUE": ["MEMORY_PER_JOB"],
     "LOCAL": [],
 }
 
+
+@dataclass
+class QueueMemoryStringFormat:
+    suffixes: List[str]
+    parse: Callable[[str], str]
+
+    def validate(self, mem_str_format: str):
+        return len([s for s in self.suffixes if mem_str_format.endswith(s)]) > 0
+
+
+def parse_lsf_memopt(s: str):
+    return s.upper()
+
+
+def parse_slurm_memopt(s: str):
+    return s.lower().replace("b", "").upper()
+
+
+def parse_torque_memopt(s: str):
+    if re.match(r"\d+[kgmt](?!\w)", s, re.IGNORECASE):
+        return s.lower() + "b"
+    if re.match(r"^\d+$", s):
+        return s + "mb"
+
+    return s
+
+
+queue_memory_usage_formats: Mapping[str, QueueMemoryStringFormat] = {
+    "LSF": QueueMemoryStringFormat(
+        suffixes=[
+            "",
+            "KB",
+            "K",
+            "MB",
+            "M",
+            "GB",
+            "G",
+            "TB",
+            "T",
+            "PB",
+            "P",
+            "EB",
+            "E",
+            "ZB",
+            "Z",
+        ],
+        parse=parse_lsf_memopt,
+    ),
+    "SLURM": QueueMemoryStringFormat(
+        suffixes=["", "K", "M", "G", "T"], parse=parse_slurm_memopt
```

Comment:
I did not check these.

Reply:
Re-checked, pasted from https://slurm.schedmd.com/sbatch.html
```
--mem=<size>[units]
Specify the real memory required per node. Default units are megabytes. Different units can be specified using the suffix [K|M|G|T]. Default value is DefMemPerNode and the maximum value is MaxMemPerNode. If configured, both parameters can be seen using the scontrol show config command. This parameter would generally be used if whole nodes are allocated to jobs (SelectType=select/linear). Also see --mem-per-cpu and --mem-per-gpu. The --mem, --mem-per-cpu and --mem-per-gpu options are mutually exclusive. If --mem, --mem-per-cpu or --mem-per-gpu are specified as command line arguments, then they will take precedence over the environment.
NOTE: A memory size specification of zero is treated as a special case and grants the job access to all of the memory on each node.
```

What do we feel about this part? 
```NOTE: A memory size specification of zero is treated as a special case and grants the job access to all of the memory on each node.``` disable/disallow/warn?

Reply:
I think you have to allow it to avoid future confusion, and we have other params that work in a similar way (timouts, MAX_RUNTIME)